cmd/admin-handlers.go

```diff
@@ -244,39 +245,111 @@ func (a adminAPIHandlers) ServerInfoHandler(w http.ResponseWriter, r *http.Reque
+		_, e = GlobalKMS.UnsealKey(vaultName, sealedKey, kmsContext)
+		if e == nil {
+			perm.Decryption = "OK"
+		} else {
+			perm.Decryption = "FAIL"
+		}
 
-	serverInfo := globalNotificationSys.ServerInfo(ctx)
-	// Once we have received all the ServerInfo from peers
-	// add the local peer server info as well.
-	serverInfo = append(serverInfo, ServerInfo{
-		Addr: getHostName(r),
-		Data: &ServerInfoData{
-			StorageInfo: objectAPI.StorageInfo(ctx),
-			ConnStats:   globalConnStats.toServerConnStats(),
-			HTTPStats:   globalHTTPStats.toServerHTTPStats(),
-			Properties: ServerProperties{
-				Uptime:       UTCNow().Sub(globalBootTime),
-				Version:      Version,
-				CommitID:     CommitID,
-				DeploymentID: globalDeploymentID,
-				SQSARN:       globalNotificationSys.GetARNList(),
-				Region:       globalServerConfig.GetRegion(),
+		vaultInfo := VaultInfo{
+			Endpoint: endPoint,
+			Name:     vaultName,
+			Type:     vaultType,
+			Perm: Permission{
+				Encryption: perm.Encryption,
+				Decryption: perm.Decryption,
 			},
-		},
-	})
+		}
 
-	// Marshal API response
-	jsonBytes, err := json.Marshal(serverInfo)
-	if err != nil {
-		writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
-		return
+		// Marshal API response
+		jsonBytes, err = json.Marshal(vaultInfo)
+		if err != nil {
+			writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+			return
+		}
+
+	case "server":
+		serverInfo := globalNotificationSys.ServerInfo(ctx)
+		// Once we have received all the ServerInfo from peers
+		// add the local peer server info as well.
+		serverInfo = append(serverInfo, ServerInfo{
+			Addr: getHostName(r),
+			Data: &ServerInfoData{
+				StorageInfo: objectAPI.StorageInfo(ctx),
+				ConnStats:   globalConnStats.toServerConnStats(),
+				HTTPStats:   globalHTTPStats.toServerHTTPStats(),
+				Properties: ServerProperties{
+					Uptime:       UTCNow().Sub(globalBootTime),
+					Version:      Version,
+					CommitID:     CommitID,
+					DeploymentID: globalDeploymentID,
+					SQSARN:       globalNotificationSys.GetARNList(),
+					Region:       globalServerConfig.GetRegion(),
+				},
+			},
+		})
+
+		// Marshal API response
+		jsonBytes, err = json.Marshal(serverInfo)
+
+		if err != nil {
+			writeErrorResponseJSON(ctx, w, toAdminAPIErr(ctx, err), r.URL)
+			return
+		}
 	}
 
 	// Reply with storage information (across nodes in a
 	// distributed setup) as json.
 	writeSuccessResponseJSON(w, jsonBytes)
 }
 
+// Permission holds the info for encrypt and decrypt
+type Permission struct {
+	Encryption string `json:"encryption,omitempty"`
+	Decryption string `json:"decryption,omitempty"`
+}
+
+// VaultInfo holds vault information
+type VaultInfo struct {
+	Endpoint string     `json:"endpoint"` // The vault API endpoint as URL
+	Name     string     `json:"name"`     // The name of the encryption key-ring
+	Type     string     `json:"type"`     // The authentication type
+	Perm     Permission `json:"perm"`
+}
```

Comment:
this definition is duplicated in madmin package. Just use madmin.VaultInfo instead

Reply:
done.